engine/access/rpc/backend/backend.go

```diff
@@ -381,9 +385,21 @@ func chooseExecutionNodes(state protocol.State, executorIDs flow.IdentifierList)
 		return nil, fmt.Errorf("failed to retreive all execution IDs: %w", err)
 	}
 
-	// If there are no preferred or fixed ENs, have the default behaviour be that
-	// we just return all the executor IDs, i.e. no preferrence at all.
+	// if no executor IDs were found (since no execution receipt has been received), then return all the preferred ENs
+	// if defined or all the fixed ENs, if the preferred ENs are not defined
+	if len(executorIDs) == 0 {
```

Comment:
This is currently needed because we don't error when we have no executorIDs? Should we be erroring instead (same conversation as above)